src/ol/source/Vector.js

```diff
@@ -299,12 +299,12 @@ class VectorSource extends Source {
   addFeatureInternal(feature) {
     const featureKey = getUid(feature);
 
+    this.setupChangeEvents_(featureKey, feature);
```

Comment:
This change will leave change events registered on features that could not be added to the source (i.e. with a spatial index).